src/table.js

```diff
@@ -1053,13 +1141,12 @@ Table.prototype.mutate = function(entries, callback) {
   }
 
   function makeNextBatchRequest() {
-    var grpcOpts = {
-      service: 'Bigtable',
-      method: 'mutateRows',
-      retryOpts: {
-        currentRetryAttempt: numRequestsMade,
-      },
-    };
+    // @todo Figure out how to do this in gapic.
```

Comment:
@kolodny Just flagging this, as I'm not currently sure how to handle this. GAPIC has its own retry logic built in: https://github.com/googleapis/gax-nodejs/blob/ec0c8b0805c31d8a91ea69cb19fe50f42a38bf87/lib/gax.js#L279. Are these options enough to configure the desired behavior?

Reply:
This looks tricky to work with as is. There's always the options of not using retries from gax and just wrapping the API in our own but that should be avoided at all costs.

My main concern is what happens when gax thinks the call was successful yet there was an error with a couple mutations out of an entire batch. Consider this (parens are timeline for step):

1. Start mutation call (0ms)
1. Can't connect to service (0ms to 100ms)
1. Wait 500ms (100ms to 600ms)
1. Some mutations failed (600ms to 650ms)
1. Wait 2500ms (650ms to 3150ms)
1. Can't connect to service (3150ms to 3250ms)
1. Wait 4500ms (3250ms to 7750ms)

The tricky bit will be the wait times of step 7 after step 5. We can have an initialRetryDelayMillis but then the multiplier will be out of sync with how the retries should behave for the remaining calls. We can either go through the pass current retry attempt dance that we did for common-grpc/retry-request or just roll our own.

@stephenplusplus @sduskis thoughts?

Reply:
> There's always the options of not using retries from gax and just wrapping the API in our own but that should be avoided at all costs.

It might not actually be that bad from a performance overhead perspective. From a "consolidating dependencies" perspective, yeah, this would deduct a few points.

Also, a new discovery: https://github.com/googleapis/gax-nodejs/blob/ec0c8b0805c31d8a91ea69cb19fe50f42a38bf87/lib/streaming.js#L230

> // Right now retrying does not work with gRPC-streaming, because retryable
> // assumes an API call returns an event emitter while gRPC-streaming methods
> // return Stream.
> // TODO: support retrying

Considering this, I've updated this PR to use retry-request.